Source/WebCore/platform/audio/gstreamer/AudioDecoderGStreamer.cpp

```diff
@@ -215,42 +205,60 @@ GStreamerInternalAudioDecoder::GStreamerInternalAudioDecoder(const String& codec
         m_inputCaps = adoptGRef(gst_caps_new_simple("audio/x-raw", "format", G_TYPE_STRING, gst_audio_format_to_string(gstPcmFormat),
             "rate", G_TYPE_INT, config.sampleRate, "channels", G_TYPE_INT, config.numberOfChannels,
             "layout", G_TYPE_STRING, "interleaved", nullptr));
-        parser = "rawaudioparse";
     } else
         return;
 
-    configureAudioDecoderForHarnessing(element);
+    if (element)
+        configureAudioDecoderForHarnessing(element);
```

Comment:
How can it be that we checked that we have a factory and that factory does not create an element?

Reply:
Good point :) The patch can be much simpler indeed, I had it this way before the raw parser was interpreted as decoder.